### PR TITLE
VSCode API Debugger

### PR DESCRIPTION
**Pull request by rylew1**
## Ticket

Resolves #TODO

## Changes

- Add API debugging ability from VSCode

## Do we want to commit IDE specific files (`.vscode/launch.json`) ?
- I don't see it in the frontend template project

## Context for reviewers

- It seems you can't run `run-logs` alongside `start-debug` for some reason - the 5678 port won't be exposed

- Demo running `make start-debug`, then connect with the VSCode launcher:

https://github.com/navapbc/template-application-flask/assets/93001277/c411675c-7e32-42ae-a70c-267bc5954486

### CODE REVIEW

File: app/src/app.py [hunk 1]
```diff
@@ -34,6 +40,29 @@ def create_app() -> APIFlask:
     return app
 
 
+def is_running_in_local() -> bool:
+    print('s.getenv("FLASK_ENV") => ', os.getenv("FLASK_ENV"))
+    return os.getenv("FLASK_ENV") in ("local", "dev")
```
Comment (chouinar) [inline app/src/app.py]: The environment is just `ENVIRONMENT`. I'd also restrict it to only `local` as some projects use `dev` as a non-local environment.

File: docker-compose.debug.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+version: "3.8"
+
+# run with `docker compose -f`
+# combines ports and env vars with main docker-compose.yml mass-pfml-api service
```
Comment (chouinar) [inline docker-compose.debug.yml]:
```suggestion
# combines ports and env vars with main docker-compose.yml main-app service
```

File: app/src/app.py [hunk 3]
```diff
@@ -21,6 +22,11 @@
 def create_app() -> APIFlask:
     app = APIFlask(__name__)
 
+    print("app => ", app)
+
+    if is_running_in_local():
+        setup_debug_adapter()
```
Comment (chouinar) [inline app/src/app.py]: We might want to put this `is_running_in_local` function in `local.py`.

https://github.com/navapbc/template-application-flask/blob/main/app/src/util/local.py

File: app/Makefile [hunk 4]
```diff
@@ -92,6 +92,9 @@ build:
 start:
 	docker compose up --detach
 
+start-debug:
+	docker compose -f ../docker-compose.yml -f ../docker-compose.debug.yml up --detach
+
```
Comment (rylew1) [inline app/Makefile:97]: debug file gets launched second to be more clear that it overrides value in the normal docker compose

**Review by chouinar — APPROVED** [review]

**Pull request merged**